jina/peapods/jinad.py

```diff
@@ -0,0 +1,193 @@
+import uuid
+from pathlib import Path
+from argparse import Namespace
+from typing import Dict, Tuple, Set
+from contextlib import ExitStack
+
+import ruamel.yaml
+
+from jina.helper import colored
+
+
+def _add_file_to_list(_file: str, _file_list: Set, logger):
+    if _file and _file.endswith(('yml', 'yaml', 'py')):
+        if Path(_file).is_file():
+            _file_list.add(_file)
+            logger.debug(f'adding file {_file} to be uploaded to remote context')
+        else:
+            logger.debug(f'file {_file} doesn\'t exist in the disk')
+
+
+def _add_files_in_main_yaml(current_pea: Dict, uses_files: Set, pymodules_files: Set, logger):
+    for _arg in ['uses', 'uses_before', 'uses_after']:
+        if _arg in current_pea:
+            _add_file_to_list(_file=current_pea[_arg],
+                              _file_list=uses_files,
+                              logger=logger)
+
+    _add_file_to_list(_file=current_pea['py_modules'],
+                      _file_list=pymodules_files,
+                      logger=logger)
+
+
+def fetch_files_from_yaml(pea_args: Dict, logger) -> Tuple[set, set]:
+    """ helper function to fetch yaml & pymodules to be uploaded to remote """
+    uses_files = set()
+    pymodules_files = set()
+
+    # This is for remote Pods
+    if 'peas' in pea_args and isinstance(pea_args['peas'], list):
+        for current_pea in pea_args['peas']:
+            _add_files_in_main_yaml(current_pea=current_pea,
+                                    uses_files=uses_files,
+                                    pymodules_files=pymodules_files,
+                                    logger=logger)
+
+    # This is for remote Peas
+    if 'peas' not in pea_args:
+        _add_files_in_main_yaml(current_pea=pea_args,
+                                uses_files=uses_files,
+                                pymodules_files=pymodules_files,
+                                logger=logger)
+
+    if uses_files:
+        for current_file in uses_files:
+            with open(current_file) as f:
+                result = ruamel.yaml.round_trip_load(f)
+
+            if 'metas' in result and 'py_modules' in result['metas']:
+                _add_file_to_list(_file=result['metas']['py_modules'],
+                                  _file_list=pymodules_files,
+                                  logger=logger)
```

Comment:
```suggestion
            _add_file_to_list(_file=result.get('metas', {}).get('py_modules'),
                              _file_list=pymodules_files,
                              logger=logger)
```

Reply:
I would say, the 1st one looks simpler & easier to handle.